### PR TITLE
Multi node prepare

### PR DESCRIPTION
**Pull request by Delaunay**

### CODE REVIEW

**Comment by Delaunay** [comment]
* Moves the execution logic out of the Pack.
* Pack is in charge of producing the command to be executed but does not execute them
* MultiPack now creates an execution plan for each pack it executes
* The execution plan launch commands on multiple nodes if available

**Comment by satyaog** [comment]
You might be a little interesting in #149 for the config and accelerate_opt benchfile.py

**Comment by Delaunay** [comment]
Remaining issue; `install` is usually a python script and not a command.
We will need to bootstrap milabench on the other nodes so we can execute the script remotely there

File: milabench/executors.py [hunk 1]
```diff
@@ -376,16 +377,16 @@ def _find_node_config(self) -> Dict:
...
+            # No-op when executing on the main node or locally
             return []
 
         node = self._find_node_config()
         user = self.user or node.get("user", None)
         key = self.key or node.get("key", None)
         host = f"{user}@{self.host}" if user else self.host
 
-        argv = super().argv(**kwargs)
+        argv = [] # super().argv(**kwargs)
```
Comment (satyaog) [inline milabench/executors.py:389]:
```suggestion
        argv = super()._argv(**kwargs)
```

Comment (satyaog) [reply milabench/executors.py:389]: The same would need to be done in `DockerRunExecutor`

**Comment by Delaunay** [comment]
Another option is simply to clone milabench on each remote nodes and
make calls to milabench itself to prepare and install so we do not need to use executor for that part of the code

**Pull request closed without merging**